[DOCS] Remove all mentions of profilers in public docs (#8825)
great-expectations/great_expectations: docs/docusaurus/docs/terms/expectation.md

## Relationship to other objects

> Expectations are grouped into <TechnicalTag relative="../" tag="expectation_suite" text="Expectation Suites" />, which are in turn stored and retrieved through an <TechnicalTag relative="../" tag="expectation_store" text="Expectation Store" />. <TechnicalTag relative="../" tag="checkpoint" text="Checkpoints" /> rely on Expectation Suites (and the Expectations contained therein) to <TechnicalTag relative="../" tag="validation" text="Validate" /> data.

Reviewer: ```suggestion
Expectations are grouped into <TechnicalTag relative="../" tag="expectation_suite" text="Expectation Suites" />, and are stored in an <TechnicalTag relative="../" tag="expectation_store" text="Expectation Store" />. <TechnicalTag relative="../" tag="checkpoint" text="Checkpoints" /> rely on Expectation Suites to <TechnicalTag relative="../" tag="validation" text="Validate" /> data.
```

There are several paths to generating Expectations:

> - Automated inspection of datasets. Currently, our Data Assistants mechanism in Great Expectations produces Expectation Suites that can be used for validation. The goal is automated inspection which can produce Expectations that will be used in validating future batches of data.

Reviewer: ```suggestion
- Automated inspection of datasets. Data Assistants produce Expectation Suites for data validation. The goal is automated inspection, which can produce Expectations that can be used to validate future batches of data.
```
